sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorModelFactory.cs

```diff
@@ -0,0 +1,477 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.AI.MetricsAdvisor.Administration;
+
+namespace Azure.AI.MetricsAdvisor.Models
+{
+    /// <summary>
+    /// A factory that builds Azure.AI.MetricsAdvisor and Azure.AI.MetricsAdvisor.Administration
+    /// model types used for mocking.
+    /// </summary>
+    public static class MetricsAdvisorModelFactory
```

Comment:
Decided to not rely on the internal model factory for two reasons:
- Autogenerated docstrings are based on the swagger file, and are not good.
- Lots of models were not included in the autogenerated factory. I think one of these two (or both) factors are responsible for this:
  - These models have internal-type inputs, and we need to make a transform from public type to internal type in the method. CodeGen is probably just skipping these, since there's not much it can do.
  - Only methods for models in the `Models` namespace are generated.